examples/07_end_to_end/seismic_waveform.ipynb

```diff
@@ -0,0 +1,1322 @@
+    "\n",
+    "Furthermore, note that because we started the csp graph 60 seconds in the past, the server has sent us a number of past observations at the start of our run, before sending us the live data. This can be useful behavior when running stateful analytics (such as moving averages) that need to be \"warmed up\"."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "84db617b-e92f-4177-9ee9-16129eae4b60",
+   "metadata": {},
+   "source": [
+    "### Real-time Triggering\n",
+    "To illustrate how the filtering/triggering logic would apply in real-time, we compute the trigger characteristic function for the \"HHZ\" channel. To limit the amount of output, instead of outputting every tick, we compute the max over regular intervals and output that. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 26,
+   "id": "95c4bbfc-2b69-4f34-9d83-4abb50f787a6",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@csp.graph\n",
+    "def rt_graph_trigger(address: str, symbols: List[Tuple[str, str, str]], nsta: int, nlta: int):\n",
+    "    adapter_manager = EasySeedLinkAdapter(address)\n",
+    "    for symbol in symbols:\n",
+    "        data = adapter_manager.subscribe(*symbol)\n",
+    "        csp.log(logging.INFO, f\"Raw data:\", data)\n",
+    "        tr_filt = bandpass_filter_ts(data, freqmin=1, freqmax=20)\n",
+    "        cft = sta_lta_ts(tr_filt, nsta, nlta)\n",
+    "        # Because on each tick of \"data\", we generate several values of \"cft\" by unrolling the observations, we want to trigger reporting of the maxim cft over each batch\n",
+    "        trigger = csp.delay(data, timedelta.resolution)\n",
+    "        min_cft = csp.stats.min(cft, trigger=trigger, reset=data)\n",
+    "        max_cft = csp.stats.max(cft, trigger=trigger, reset=data)        \n",
+    "        csp.log(logging.INFO, f\"Characteristic function for {symbol}, Min\", min_cft)\n",
+    "        csp.log(logging.INFO, f\"Characteristic function for {symbol}, Max\", max_cft)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a9df9119-4050-4a90-94f1-53ee0fe9b197",
+   "metadata": {},
+   "source": [
+    "For this example, we use a station on the Canadian (i.e. \"CN\") network, as the high frequency channel will publish updates every 4-5 seconds, which is more rapid than the IRIS network used above. However, occasionally the connection to this network can time out, in which case, please try again later. When it's working properly, a log line containing `INFO: network socket opened` should appear almost immediately.\n",
+    "The [GEOFON network](https://geofon.gfz-potsdam.de/waveform/seedlink.php) (http://geofon.gfz-potsdam.de:18000) is also a good alternative choice, i.e. channel (\"GE\", \"STU\", \"HHZ\"). "
```

Comment:
The Canadian network was timed out when I tried, but I used the German one and it worked.
Maybe we should just use the German one for this example and mention the Canadian one as an alternative for people top try out?

Reply:
Not allowed to for compliance reasons.